File: LeetCode/1822. Sign of the Product of an Array/solution.py
"""
Runtime: 36 ms
Memory Usage: 15.1 MB
"""
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        neg = 0
        for num in nums:
            if num == 0:
                return 0
            if num < 0:
                neg += 1
        return 1 if neg & 1 == 0 else -1
        

File: LeetCode/1822. Sign of the Product of an Array/Solution.java
/**
 * Runtime: 0 ms
 * Memory Usage: 40.9 MB 
 */
class Solution {
    public int arraySign(int[] nums) {
        int sign = 1;
        for (int num: nums) {
            if (num == 0) return 0;
            if (num < 0) sign *= -1;
        }
        return sign;
    }
}
